EarlGrey/Exception/GREYDefaultFailureHandler.m

```diff
@@ -52,11 +52,12 @@ - (void)handleException:(GREYFrameworkException *)exception details:(NSString *)
   }
 
   NSMutableString *log = [[NSMutableString alloc] init];
-  // Start on fresh new line.
-  [log appendString:@"\n"];
-  [log appendFormat:@"Exception: %@\n", exception.name];
   NSString *reason =
       exception.reason.length > 0 ? exception.reason : @"exception.reason was not provided";
+  [log appendFormat:@"%@\n\n", reason];
+  [log appendFormat:@"Bundle ID: %@\n\n", [[NSBundle mainBundle] bundleIdentifier]];
```

Comment:
Let's keep it.

In cases where someone copy and pastes the test log or a part of the test log it may not be clear which app bundle the snippet is from, especially when the development and production copy of the app have a different bundle ID so you can have both on your phone at the same time.

Furthermore, in the multiprocess model log output needs to have a bundle ID so you know which app generated the error message.